src/config/mongoose_config_parser_toml.erl

```diff
@@ -518,30 +519,707 @@ process_module([Mod|_] = Path, Opts) ->
+    [{secret_access_key, b2l(V)}];
+mod_event_pusher_backend_sns_opts([<<"account_id">>|_], V) ->
+    [{account_id, b2l(V)}];
+mod_event_pusher_backend_sns_opts([<<"pool_size">>|_], V) ->
+    [{pool_size, V}];
+mod_event_pusher_backend_sns_opts([<<"publish_retry_count">>|_], V) ->
+    [{publish_retry_count, V}];
+mod_event_pusher_backend_sns_opts([<<"publish_retry_time_ms">>|_], V) ->
+    [{publish_retry_time_ms, V}].
+
+-spec mod_event_pusher_backend_push_opts(path(), toml_value()) -> [option()].
+mod_event_pusher_backend_push_opts([<<"backend">>|_], V) ->
+    [{backend, b2a(V)}];
+mod_event_pusher_backend_push_opts([<<"wpool">>|_] = Path, V) ->
+    WpoolOpts = parse_section(Path, V),
+    [{wpool, WpoolOpts}];
+mod_event_pusher_backend_push_opts([<<"plugin_module">>|_], V) ->
+    [{plugin_module, b2a(V)}];
+mod_event_pusher_backend_push_opts([<<"virtual_pubsub_hosts">> |_] = Path, V) ->
+    VPH = parse_list(Path, V),
+    [{virtual_pubsub_hosts, VPH}].
+
+-spec mod_event_pusher_backend_http_opts(path(), toml_value()) -> [option()].
+mod_event_pusher_backend_http_opts([<<"pool_name">>|_], V) ->
+    [{pool_name, b2a(V)}];
+mod_event_pusher_backend_http_opts([<<"path">>|_], V) ->
+    [{path, b2l(V)}];
+mod_event_pusher_backend_http_opts([<<"callback_module">>|_], V) ->
+    [{callback_module, b2a(V)}].
+
+-spec mod_event_pusher_backend_rabbit_opts(path(), toml_value()) -> [option()].
+mod_event_pusher_backend_rabbit_opts([<<"presence_exchange">>|_] = Path, V) ->
+    [{presence_exchange, parse_section(Path, V)}];
+mod_event_pusher_backend_rabbit_opts([<<"chat_msg_exchange">>|_] = Path, V) ->
+    [{chat_msg_exchange, parse_section(Path, V)}];
+mod_event_pusher_backend_rabbit_opts([<<"groupchat_msg_exchange">>|_] = Path, V) ->
+    [{groupchat_msg_exchange, parse_section(Path, V)}].
+
+-spec mod_event_pusher_rabbit_presence_ex(path(), toml_value()) -> [option()].
+mod_event_pusher_rabbit_presence_ex([<<"name">>|_], V) ->
+    [{name, V}];
+mod_event_pusher_rabbit_presence_ex([<<"type">>|_], V) ->
+    [{type, V}].
+
+-spec mod_event_pusher_rabbit_msg_ex(path(), toml_value()) -> [option()].
+mod_event_pusher_rabbit_msg_ex([<<"name">>|_], V) ->
+    [{name, V}];
+mod_event_pusher_rabbit_msg_ex([<<"type">>|_], V) ->
+    [{type, V}];
+mod_event_pusher_rabbit_msg_ex([<<"sent_topic">>|_], V) ->
+    [{sent_topic, V}];
+mod_event_pusher_rabbit_msg_ex([<<"recv_topic">>|_], V) ->
+    [{recv_topic, V}].
+
+-spec mod_http_upload_s3(path(), toml_value()) -> [option()].
+mod_http_upload_s3([<<"bucket_url">>|_], V) ->
+    [{bucket_url, b2l(V)}];
+mod_http_upload_s3([<<"add_acl">>|_], V) ->
+    [{add_acl, V}];
+mod_http_upload_s3([<<"region">>|_], V) ->
+    [{region, b2l(V)}];
+mod_http_upload_s3([<<"access_key_id">>|_], V) ->
+    [{access_key_id, b2l(V)}];
+mod_http_upload_s3([<<"secret_access_key">>|_], V) ->
+    [{secret_access_key, b2l(V)}].
+
+-spec mod_global_distrib_connections(path(), toml_section()) -> [option()].
```

Comment:
Here it should be `toml_value()` - it is the supertype of `toml_section()`, not otherwise.